packages/dashboard-frontend/src/preload/index.ts

```diff
@@ -10,10 +10,15 @@
  *   Red Hat, Inc. - initial API and implementation
  */
 
+import SessionStorageService, { SessionStorageKey } from '../services/session-storage';
+
 (function acceptNewFactoryLink(): void {
   if (window.location.pathname.startsWith('/dashboard/')) {
     return;
   }
+
+  SessionStorageService.update(SessionStorageKey.ORIGINAL_LOCATION_PATH, window.location.pathname);
```

Comment:
When a workspace is idled and redirected to the dashboard by this PR: https://github.com/eclipse/che/issues/20599, `window.location.pathname` would be something like: `/workspacee3dab47b6f6441c6/theia-ide/3100/#/`

Reply:
I think it should include some checks:
```
...
  const pathname = window.location.pathname;
  if (pathname !== '/') {
    SessionStorageService.update(SessionStorageKey.ORIGINAL_LOCATION_PATH, pathname);
  }
  ...
```